Graphics/Slicer/Machine/Infill.hs

```diff
@@ -71,30 +71,26 @@ infillLineSegInside contour childContours line
           filterTooShort [a] = [a]
           filterTooShort (a:b:xs) = if roundToFifth (distance a b) == 0 then filterTooShort xs else a:filterTooShort (b:xs)
           getLineSegIntersections :: LineSeg -> Contour -> [Point2]
-          getLineSegIntersections myline c = saneIntersections $ lineIntersection myline <$> linesOfContour c
+          getLineSegIntersections myline c = saneIntersections $ intersectsWith (Left myline) . Left <$> linesOfContour c
             where
               linesOfContour (PointSequence contourPoints) = makeLineSegsLooped $ (\(Point2 (x,y)) -> Point2 (roundToFifth x, roundToFifth y)) <$> contourPoints
-              saneIntersections :: [Intersection] -> [Point2]
-              saneIntersections xs = concat $ mapMaybe saneIntersection xs
+              saneIntersections :: [Either Intersection PIntersection] -> [Point2]
+              saneIntersections xs = catMaybes $ mapWithNeighbors saneIntersection xs
                 where
-                  saneIntersection :: Intersection -> Maybe [Point2]
+                  saneIntersection :: Either Intersection PIntersection -> Either Intersection PIntersection -> Either Intersection PIntersection -> Maybe Point2
+                  saneIntersection _ (Right (IntersectsIn ppoint)) _ = Just $ pToEPoint2 ppoint
+                  saneIntersection _ (Left NoIntersection)         _ = Nothing
+                  saneIntersection _ (Right PParallel)             _ = Nothing
+                  saneIntersection _ (Right PAntiParallel)         _ = Nothing
+                  saneIntersection _                          (Left (HitStartPointL2 point)) (Left (HitEndPointL2   _)) = Just point
+                  saneIntersection (Left (HitStartPointL2 _)) (Left (HitEndPointL2   _    )) _                          = Nothing
+                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Left (HitStartPointL2 _)) = Nothing
+                  saneIntersection (Left (HitEndPointL2   _)) (Left (HitStartPointL2 point)) _                          = Just point
+                  saneIntersection (Right PColinear)          (Left (HitStartPointL2 _    )) _                          = Nothing
+                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Right PColinear)          = Nothing
+                  saneIntersection _                          (Right PColinear)              _                          = Nothing
+                  saneIntersection r1 r2 r3 = error $ "insane result of intersecting a line (" <> show myline <> ") with a contour " <> show c <> "\n" <> show r1 <> "\n" <> show r2 <> "\n" <> show r3 <> "\n"
```

Comment:
Avoid `error` calls. They only bring pain and sadness.

Reply:
They should, in this case. i'm not done handling all of the intersections here.

Reply:
A question I ask when reviewing PRs professionally --- "if you don't have time to do it correctly now, when will you?"